river/feature_extraction/vectorize.py

```diff
@@ -259,6 +271,28 @@ class BagOfWords(base.Transformer, VectorizerMixin):
     def transform_one(self, x):
         return collections.Counter(self.process_text(x))
 
+    def transform_many(self, X: pd.Series):
+        """Transform pandas series of string."""
+        X_p = [list(self.process_text(x)) for x in X]
+
+        idx, i = {}, 0
+        for x_p in X_p:
+            for x in x_p:
+                if x not in idx:
+                    idx[x] = i
+                    i += 1
+
+        array = np.zeros((len(X_p), len(idx)), dtype=int)
+
+        for row, x in enumerate(X_p):
+            for col in x:
+                array[row][idx[col]] += 1
+
+        return pd.DataFrame(array, columns=idx.keys(), index=X.index)
```

Comment:
Interesting! Could you try out using [scipy's `lil_matrix`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.lil_matrix.html)? You might be able to kill two birds with one stone:

1. Possibly.
2. You don't need to do numpy -> pandas full -> pandas sparse (which could save time and memory). The problem with doing `bag.transform_many(X).astype('Sparse[int]')` is not just the time overhead: it incurs a memory spike that we would like to avoid.